pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -915,6 +925,46 @@ const (
 	LegacyNameValidationScheme NameValidationSchemeOptions = "Legacy"
 )
 
+// Specifies the character escaping scheme that will be requested when scraping
+// for metric and label names that do not conform to the legacy Prometheus
+// character set.
+//
+// Supported values are:
+//   - `AllowUTF8NameEscapingScheme` for Full UTF-8 support, no escaping needed.
+//   - `UnderscoresNameEscapingScheme` for Escape all legacy-invalid characters to underscores.
+//   - `DotsNameEscapingScheme` for Escapes dots to `_dot_`, underscores to `__`, and all other
+//     legacy-invalid characters to underscores.
+//   - `ValuesNameEscapingScheme` for Prepend the name with `U__` and replace all invalid
+//     characters with their unicode value, surrounded by underscores.
+//
+// +kubebuilder:validation:Enum=AllowUTF8;Underscores;Dots;Values
+type NameEscapingSchemeOptions string
+
+const (
+	AllowUTF8NameEscapingScheme   NameEscapingSchemeOptions = "AllowUTF8"
+	UnderscoresNameEscapingScheme NameEscapingSchemeOptions = "Underscores"
+	DotsNameEscapingScheme        NameEscapingSchemeOptions = "Dots"
+	ValuesNameEscapingScheme      NameEscapingSchemeOptions = "Values"
+)
+
+func (s *NameEscapingSchemeOptions) ToString() string {
```

Comment:
(suggestion) I'd move the conversion to pkg/prometheus/promcfg.go.

Reply:
Thanks @simonpasquier 
Got it